### PR TITLE
Interface for AFNI's 3dBlurToFWHM

### PR DESCRIPTION
**Pull request by TheChymera**
new interface, also corrected minor typo

### CODE REVIEW

File: nipype/interfaces/afni/preprocess.py [hunk 1]
```diff
@@ -21,6 +21,37 @@
...
+    """Blurs a 'master' dataset until it reaches a specified FWHM smoothness (approximately).
+
+    For complete details, see the `to3d Documentation
+    <https://afni.nimh.nih.gov/pub/dist/doc/program_help/3dBlurToFWHM.html>`_
+
+    Examples
+    ========
+
+    >>> from nipype.interfaces import afni
+    >>> blur = afni.preprocess.BlurToFWHM()
```
Comment (oesteban) [inline nipype/interfaces/afni/preprocess.py]: Please add ```BlurToFWHM``` to the ```__init__.py``` in afni's folder. This way you can remove that ```preprocess``` after afni

**Comment by satra** [comment]
@TheChymera - please run `make specs` as well and add the `test_auto_BlurToFWHM` file.

**Comment by TheChymera** [comment]
@satra I ran `make specs` in the nipy root. But it seems to have introduced only changes entirely unrelated to `bru2nii`:

```
chymera@quiethost ~/src/nipype $ make specs
Checking specs and autogenerating spec tests
python tools/checkspecs.py
160617-14:41:56,806 interface WARNING:
	 VTK was not found
160617-14:41:56,807 interface WARNING:
	 tvtk wasn't found
FILMGLS has nonautotest
Level1Design has nonautotest
/usr/lib64/python2.7/site-packages/nipy/labs/glm/glm.py:9: FutureWarning: Module nipy.labs.utils.routines deprecated, will be removed
  from ..utils import mahalanobis
/usr/lib64/python2.7/site-packages/nipype/interfaces/nipy/model.py:18: FutureWarning: Module nipy.labs.glm deprecated, will be removed. Please use nipy.modalities.fmri.glm instead.
  import nipy.labs.glm.glm as GLM
```


And this is the output of `git diff` after I run the above:

```
diff --git a/nipype/interfaces/freesurfer/tests/test_auto_Sphere.py b/nipype/interfaces/freesurfer/tests/test_auto_Sphere.py
index f66f910..ceab66d 100644
--- a/nipype/interfaces/freesurfer/tests/test_auto_Sphere.py
+++ b/nipype/interfaces/freesurfer/tests/test_auto_Sphere.py
@@ -20,6 +20,7 @@ def test_Sphere_inputs():
     in_smoothwm=dict(copyfile=True,
     ),
     magic=dict(argstr='-q',
+    requires=['in_smoothwm'],
     ),
     num_threads=dict(),
     out_file=dict(argstr='%s',
diff --git a/nipype/interfaces/fsl/tests/test_auto_ApplyTOPUP.py b/nipype/interfaces/fsl/tests/test_auto_ApplyTOPUP.py
index f7a8f49..77a11e3 100644
--- a/nipype/interfaces/fsl/tests/test_auto_ApplyTOPUP.py
+++ b/nipype/interfaces/fsl/tests/test_auto_ApplyTOPUP.py
@@ -22,6 +22,7 @@ def test_ApplyTOPUP_inputs():
     sep=',',
     ),
     in_index=dict(argstr='--inindex=%s',
+    mandatory=True,
     sep=',',
     ),
     in_topup_fieldcoef=dict(argstr='--topup=%s',
diff --git a/nipype/interfaces/fsl/tests/test_auto_BEDPOSTX5.py b/nipype/interfaces/fsl/tests/test_auto_BEDPOSTX5.py
index 9b02366..48f229e 100644
--- a/nipype/interfaces/fsl/tests/test_auto_BEDPOSTX5.py
+++ b/nipype/interfaces/fsl/tests/test_auto_BEDPOSTX5.py
@@ -36,7 +36,6 @@ def test_BEDPOSTX5_inputs():
     ),
     fudge=dict(argstr='-w %d',
     ),
-    grad_dev=dict(),
     gradnonlin=dict(argstr='-g',
     ),
     ignore_exception=dict(nohash=True,
diff --git a/nipype/interfaces/fsl/tests/test_auto_Level1Design.py b/nipype/interfaces/fsl/tests/test_auto_Level1Design.py
deleted file mode 100644
index f1500d4..0000000
--- a/nipype/interfaces/fsl/tests/test_auto_Level1Design.py
+++ /dev/null
@@ -1,35 +0,0 @@
-# AUTO-GENERATED by tools/checkspecs.py - DO NOT EDIT
-from ....testing import assert_equal
-from ..model import Level1Design
-
-
-def test_Level1Design_inputs():
-    input_map = dict(bases=dict(mandatory=True,
-    ),
-    contrasts=dict(),
-    ignore_exception=dict(nohash=True,
-    usedefault=True,
-    ),
-    interscan_interval=dict(mandatory=True,
-    ),
-    model_serial_correlations=dict(mandatory=True,
-    ),
-    session_info=dict(mandatory=True,
-    ),
-    )
-    inputs = Level1Design.input_spec()
-
-    for key, metadata in list(input_map.items()):
-        for metakey, value in list(metadata.items()):
-            yield assert_equal, getattr(inputs.traits()[key], metakey), value
-
-
-def test_Level1Design_outputs():
-    output_map = dict(ev_files=dict(),
-    fsf_files=dict(),
-    )
-    outputs = Level1Design.output_spec()
-
-    for key, metadata in list(output_map.items()):
-        for metakey, value in list(metadata.items()):
-            yield assert_equal, getattr(outputs.traits()[key], metakey), value
diff --git a/nipype/interfaces/semtools/diffusion/tests/test_auto_DWIConvert.py b/nipype/interfaces/semtools/diffusion/tests/test_auto_DWIConvert.py
index 556f2b0..15e293f 100644
--- a/nipype/interfaces/semtools/diffusion/tests/test_auto_DWIConvert.py
+++ b/nipype/interfaces/semtools/diffusion/tests/test_auto_DWIConvert.py
@@ -51,10 +51,6 @@ def test_DWIConvert_inputs():
     ),
     writeProtocolGradientsFile=dict(argstr='--writeProtocolGradientsFile ',
     ),
-    allowLossyConversion=dict(argstr='--allowLossyConversion ',
-    ),
-    transposeInputBVectors=dict(argstr='--transposeInputBVectors ',
-    ),
     )
     inputs = DWIConvert.input_spec()
 ```

**Comment by satra** [comment]
@TheChymera - that is weird. let me check and push a clean master.

**Comment by TheChymera** [comment]
@satra set me know when I should rebase.

**Comment by satra** [comment]
@TheChymera - you should be able to merge master and then do `make specs`

**Comment by TheChymera** [comment]
@satra Ok, I rebased, but again, `make specs` only effects changes unrelated to my commits:

```
diff --git a/nipype/interfaces/afni/tests/test_auto_DegreeCentrality.py b/nipype/interfaces/afni/tests/test_auto_DegreeCentrality.py
index bdcf111..36f446c 100644
--- a/nipype/interfaces/afni/tests/test_auto_DegreeCentrality.py
+++ b/nipype/interfaces/afni/tests/test_auto_DegreeCentrality.py
@@ -24,6 +24,7 @@ def test_DegreeCentrality_inputs():
     mask=dict(argstr='-mask %s',
     ),
     oned_file=dict(argstr='-out1D %s',
+    mandatory=False,
     ),
     out_file=dict(argstr='-prefix %s',
     name_source=['in_file'],
diff --git a/nipype/interfaces/afni/tests/test_auto_Seg.py b/nipype/interfaces/afni/tests/test_auto_Seg.py
index 753e2b0..3a84f00 100644
--- a/nipype/interfaces/afni/tests/test_auto_Seg.py
+++ b/nipype/interfaces/afni/tests/test_auto_Seg.py
@@ -30,6 +30,7 @@ def test_Seg_inputs():
     main_N=dict(argstr='-main_N %d',
     ),
     mask=dict(argstr='-mask %s',
+    exists=True,
     mandatory=True,
     position=-2,
     ),
diff --git a/nipype/interfaces/freesurfer/tests/test_auto_MRIsConvert.py b/nipype/interfaces/freesurfer/tests/test_auto_MRIsConvert.py
index d200a1e..15ac4b2 100644
--- a/nipype/interfaces/freesurfer/tests/test_auto_MRIsConvert.py
+++ b/nipype/interfaces/freesurfer/tests/test_auto_MRIsConvert.py
@@ -56,7 +56,7 @@ def test_MRIsConvert_inputs():
     ),
     to_scanner=dict(argstr='--to-scanner',
     ),
-    to_tkr=dict(argstr='--to-tkr',
+    to_tkr=dict(agrstr='--to-tkr',
     ),
     vertex=dict(argstr='-v',
     ),
```

**Pull request merged**